src/js/components/DSLInputField.js

```diff
@@ -72,7 +91,21 @@ class DSLInputField extends React.Component {
    * @param {SyntheticEvent} event - The change event
    */
   handleChange({target}) {
-    this.props.onChange(new DSLExpression(target.value));
+    this.setState({
+      expression: new DSLExpression(target.value)
+    });
+
+    clearTimeout(this.debounceTimer);
+    this.debounceTimer = setTimeout(
+      this.handleDebounceUpdate, DEBOUNCE_TIMEOUT
+    );
+  }
+
+  /**
+   * Forward property change update after the debounce timer expired
+   */
+  handleDebounceUpdate() {
```

Comment:
I had the same thinking. Having a unique ID to somehow identify your de-bounce timer ID. That would actually be perfect. But this introduces a new concept that I would like to run through the team first.